client/api_bucket.go

```diff
@@ -154,11 +154,18 @@ func (c *client) CreateBucket(ctx context.Context, bucketName string, primaryAdd
 		return "", err
 	}
 
+	ctxTimeout, cancel := context.WithTimeout(ctx, types.ContextTimeout)
+	defer cancel()
+
+	var txnResponse *sdk.TxResponse
 	txnHash := resp.TxResponse.TxHash
 	if !opts.IsAsyncMode {
-		_, err = c.WaitForTx(ctx, txnHash)
+		txnResponse, err = c.WaitForTx(ctxTimeout, txnHash)
 		if err != nil {
-			return txnHash, errors.New("failed to commit txn:" + err.Error())
+			return txnHash, fmt.Errorf("the transaction has been submitted, please check it later:%v", err)
+		}
+		if txnResponse.Code != 0 {
+			return txnHash, fmt.Errorf("the createBucket txn has failed with response code: %d", txnResponse.Code)
```

Comment:
Did we have any plan to support retry in sdk API?

Reply:
Not yet,  we can evaluate which scenarios need to be retried and support it later.